pkg/controllers/vdb/sandboxsubcluster_reconciler.go

```diff
@@ -455,11 +462,20 @@ func (s *SandboxSubclusterReconciler) addSandboxedSubclusterToStatus(ctx context
 }
 
 func (s *SandboxSubclusterReconciler) waitForSandboxScUp(ctx context.Context, sb, sc string) (ctrl.Result, error) {
-	initiatorIPs, res, err := s.findInitiatorIPs(ctx, sb)
-	if verrors.IsReconcileAborted(res, err) {
-		return res, err
+	pfs := s.PFacts.Copy(sb)
+	if err := pfs.Collect(ctx, s.Vdb); err != nil {
```

Comment:
Collect pod facts is expensive. We should collect it once for all subclusters in the sandbox. We should move it outside this function or change the argument `sc string` to `scs []string`. And move the line 216 below line 220.